tutorials/arvo/cli-tutorial.md

```diff
@@ -0,0 +1,420 @@
+instantly run the command. If it's false, require the user to press return.
+
+#### `+tab-list`
+
+```hoon
+  ++  tab-list
+    |~  sole-id=@ta
+    *(list (option:auto tank))
+```
+
+Autocomplete options for the command-line session (to match `+command-parser`).
+
+#### `+on-command`
+
+```hoon
+  ++  on-command
+    |~  [sole-id=@ta command=command-type]
+    *(quip card _this)
+```
+
+Called when a valid command is run.
+
+#### `+can-connect`
+
+```hoon
+  ++  can-connect
+    |~  sole-id=@ta
+    *?
+```
+
+Called to determine whether a session may be opened or connected to. For example, you
+may only want the local ship to be able to connect.
+
+#### `+on-connect`
+
+```hoon
+  ++  on-connect
+    |~  sole-id=@ta
+    *(quip card _^|(..on-init))
+```
+
+Called when a session is opened or connected to.
+
+#### `+on-disconnect`
+
+```hoon
+  ++  on-disconnect
+    |~  sole-id=@ta
+    *(quip card _^|(..on-init))
+```
+
+Called when a previously made session gets disconnected from.
+
+### `default` core
+
+This core contains the bare minimum implementation of the additional `shoe` arms
+beyond the 10 standard Gall app ams. It is used
+analogously to how the `default-agent` core is used for regular Gall apps.
+
+### `agent` core
+
+This is a function for wrapping a `shoe` core, which has too many
+arms to be a valid Gall agent core. This turns it into a standard Gall agent core by
+integrating the additional arms into the standard ones.
+
+
+## The `sole` library {#the-sole-library}
+
+`shoe` apps may create specialized `card`s of the `[%shoe
+(list @ta) shoe-effect]` shape, where `shoe-effect` currently just wrap `sole-effect`s, i.e. instructions for displaying text and producing other effects in the console.
+
+The list of possible `sole-effects` can be found in `/sur/sole.hoon`. A few
+commonly used ones are as follows.
+- `[%txt tape]` is used to display a line of text.
+- `[%bel ~]` is used to emit a beep.
+- `[%pro sole-prompt]` is used to set the prompt.
+- `[%mor (list sole-effect)]` is used to emit multiple effects.
+
+For example, a `sole-effect` that beeps and displays `This is some text.` would
+be structured as
+```hoon
+[%mor [%txt "This is some text."] [%bel ~] ~]
+```
+
+## `%shoe` app walkthrough {#shoe-app-walkthrough}
+
+Here we explore the capabilities of the `%shoe` example app and then go through
+the code, explaining what each line does. 
+
+### Playing with `%shoe`
+
+First let's test the functionality of `%shoe` so we know what we're getting
+into.
+
+Start two fake ships, one named `~zod` and the other can have any name - we will
+go with `~nus`. Fake ships run locally are able to see each other, and our
+intention is to connect their `%shoe` apps.
+
+On each fake ship start `%shoe` by entering `|start %shoe` into dojo. This will
+automatically
+change the prompt to `~zod:shoe>` and `~nus:shoe>`. Enter `demo` from `~zod` and press Enter:
```

Comment:
>press Enter

I was gonna say it's "press return", but, you don't need to. Part of the demo is that the command runs instantly upon typing it in. Was this not your experience?

```suggestion
change the prompt to `~zod:shoe>` and `~nus:shoe>`. Type `demo` and watch the following appear:
```

Reply:
I do have a vivid memory of needing to press Enter, and then a few days later being surprised to find that I no longer needed to. I haven't been sure of what to make of it, whether my memory is faulty, my perception was wrong, my computer was lagging, a bug somewhere, etc. But knowing that not needing to press Enter is the intended behavior is good enough for this edit at least.